src/packs/checker/architecture.rs

```diff
@@ -34,6 +35,32 @@ impl Layers {
         }
     }
 }
+
+pub fn is_architecture_dependency(
```

Comment:
What do you think of calling this `dependency_permitted?` and having it return true if it's okay?